### PR TITLE
[python] Add document for debugging chip-device-ctrl

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem

We received report that chip-device-ctrl is hard to debug.

 #### Summary of Changes

Add a guide for debugging using GDB under linux.

### CODE REVIEW

**Comment by erjiaqing** [comment]
@cjandhyala Please review the doc and notice me if something is missing or unclear, thanks.

File: src/controller/python/README.md [hunk 1]
```diff
@@ -421,3 +421,177 @@ example, `networkId=hex:0123456789abcdef` (for
...
+    <http://www.gnu.org/software/gdb/documentation/>.
+
+For help, type "help".
+Type "apropos word" to search for commands related to "word"...
+Reading symbols from python3...
+(No debugging symbols found in python3)
+(gdb)
+```
+
+The Python will create lots of threads due to main look, so you may want to
```
Comment (andy31415) [inline src/controller/python/README.md:458]: s/look/loop

**Review by andy31415 — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

File: src/controller/python/README.md [hunk 2]
```diff
@@ -421,3 +421,177 @@ example, `networkId=hex:0123456789abcdef` (for
...
+CHIP:CTL: Generating credentials
+CHIP:CTL: Loaded credentials successfully
+CHIP:DL: Platform main loop started.
+Chip Device Controller Shell
+
+chip-device-ctrl >
+```
+
+The prompt `chip-device-ctrl >` indicates that the CHIP core library is loaded
+by Python, you can browse the symbols in the CHIP core library, setting
```
Comment (Damian-Nordic) [inline src/controller/python/README.md:491]: "setting" -> "set"?

File: src/controller/python/README.md [hunk 3]
```diff
@@ -421,3 +421,177 @@ example, `networkId=hex:0123456789abcdef` (for
...
+breakpoints on functions and many other functions provided by GDB.
```
Comment (Damian-Nordic) [inline src/controller/python/README.md:492]: "and **use** many other" ?

**Review by Damian-Nordic — APPROVED** [review]

**Pull request merged**